### PR TITLE
perf: Lazy load dashboard components

### PR DESCRIPTION
**Pull request by juliushaertl**
Save loading and evaluating larger amounts of javascript on the early page load.

### CODE REVIEW

**Comment by juliushaertl** [comment]
@julien-nc Might be something nice for all the integration widgets at some point :)

**Comment by julien-nc** [comment]
:+1: 
It does not work with a subpath on my local setup. I see a request to `https://nextcloud.local/apps/notes/js/notes-commons.js?v=1fe6d2ef17a17417d45d` while it should be `https://nextcloud.local/dev/server26/apps/notes/js/notes-commons.js?v=1fe6d2ef17a17417d45d`

File: webpack.config.js [hunk 1]
```diff
@@ -3,4 +3,14 @@ const path = require('path')
 
 webpackConfig.entry.dashboard = path.join(__dirname, 'src', 'dashboard.js')
 
+webpackConfig.optimization = {
+	splitChunks: {
+		cacheGroups: {
+			defaultVendors: {
+				name: 'commons',
```
Comment (julien-nc) [inline webpack.config.js:10]:
```suggestion
				name: 'dashboard',
```
This fixed it, at least in my context. Was it a mistake or did you intentionally wrote "commons" here?

**Review by julien-nc — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 1 above)
Comment (julien-nc) [reply webpack.config.js:10]: Looking a bit at the Webpack doc and other apps, maybe this is the proper fix:
```
__webpack_nonce__ = btoa(OC.requestToken) // eslint-disable-line
__webpack_public_path__ = OC.linkTo('notes', 'js/') // eslint-disable-line
```

I don't quite get how one chooses what to put in which chunk file.

Comment (juliushaertl) [reply webpack.config.js:10]: Actually the second one would be the proper fix, sorry I missed that.

**Review by julien-nc — APPROVED** [review]
:+1:

**Pull request merged**